### PR TITLE
Redesign & Refac Fund, Campaign & Pledge Screen

### LINKED ISSUE #1927: ADMIN REDESIGN: More intuitive fundraising screens
**Is your feature request related to a problem? Please describe.**

We need more intuitive fundraising screens with updated functionality.

**Describe the solution you'd like**

1. When a fund is created:
    1. The `Manage` button updates the base fund parameters. `Edit` would be more appropriate.
    2. Clicking the fund's name, gets you to the fund's Campaigns screen. It's not clear that this is what the link does. We need a button or some other device to do this. `View Campaigns` or something similar would be better
    3. The `Non Archived` status is technically correct, but not intuitive. It should be `Active`
    4. The tab text needs to be updated
1. When clicking through on a Fund to view the campaigns:
    1. The left drawer doesn't indicate that you got to this screen via Funds. We normally use green for selected items, but this is one screen in, so possibly another color or shade would be appropriate. Suggestions welcome, I'm not familiar with the best practices on this.
    1. Creating a campaign gives an `input validation` error
        1. The newly created campaign must be visible in the list for the fund.
        2. There must be a way to edit the campaign 
1. Tests must be created for these updates. The must be valid and pass. We especially don't want this campaign failure bug occurring again.

**Describe alternatives you've considered**

- See above

**Approach to be followed (optional)**

- See above
**Additional context**

1. Funds screen:

    image

1. The tab text that needs updating:

    image

1. Campaigns screen:

    image

1. The create campaign modal:

    image

1. Create campaign `input validation` error:

    image

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by GlenDsza**
**What kind of change does this PR introduce?**

feature, redesign, refactoring

**Issue Number:**

Fixes #1927 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

Fund Screen:
image

Campaign Screen:
image

Pledge Screen:
image

Video:

https://github.com/PalisadoesFoundation/talawa-admin/assets/109027247/46c526ad-d9f4-4daa-9542-29c5d5cc3984

**Summary**

- Redesigned & Refactored (modularized) Fund Screen and its respective create/update/delete modals (popups).
- Redesigned & Refactored (modularized) Campaign Screen and its respective create/update/delete modals (popups).
- Added Breadcrumb for efficient navigation back and forth.
- Added search & sort functionality in Fund/Campaign Screen
- Added campaign progress indicator in Pledge Screen (For now static).
- Added testcases for above changes

**Does this PR introduce a breaking change?**

No

**Other information**

- The progress indicator in Pledge Screen is for now static, as currently we miss the screens/ability for user to pledge and pay for same. In the following PR this feature would be dynamically integrated.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes 

## Summary by CodeRabbit

- **New Features**
  - Introduced sorting functionality to the fund management UI.
  - Implemented `DataGrid` for displaying fund information.
  - Added `FundModal` and `FundDeleteModal` for improved fund management.

- **Improvements**
  - Refined text labels and messages across multiple languages for better clarity and consistency.
  - Enhanced campaign overview display with breadcrumb navigation and additional details.
  - Updated styles for better user interaction and visual consistency.

- **Bug Fixes**
  - Corrected translations and terminology in various languages for accuracy.

- **Refactor**
  - Restructured components and adjusted state management in `OrganizationFunds.tsx` for improved code maintainability and readability.
  - Consolidated fund modals into a single `FundModal` component.
  - Improved state management and type safety in `FundCampaignPledge.tsx`.

### CODE REVIEW

File: src/screens/OrganizationFundCampaign/CampaignDeleteModal.tsx [hunk 1]
```diff
@@ -1,29 +1,53 @@
 import React from 'react';
 import { Button, Modal } from 'react-bootstrap';
 import styles from './OrganizationFundCampaign.module.css';
+import { useMutation } from '@apollo/client';
+import { DELETE_CAMPAIGN_MUTATION } from 'GraphQl/Mutations/CampaignMutation';
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/CampaignDeleteModal.tsx:5]: Invalid Suggestion: Folder containing mutation queries is named as `GraphQl`, hence the following path is used

File: src/screens/OrganizationFundCampaign/CampaignDeleteModal.tsx [hunk 2]
```diff
@@ -1,29 +1,53 @@
...
+        variables: {
+          id: campaign?._id,
+        },
+      });
+      toast.success(t('deletedCampaign'));
+      refetchCampaign();
+      hide();
+    } catch (error: unknown) {
+      toast.error((error as Error).message);
+    }
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/CampaignDeleteModal.tsx:41]: The toast is displayed on delete Action, hence giving additional context of origin of error is not required. And error.message sent from server is enough for context

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 3]
```diff
@@ -1,77 +1,89 @@
...
+    setModalState((prevState) => ({ ...prevState, [modal]: false }));
+
+  const handleOpenModal = useCallback(
+    (fund: InterfaceFundInfo | null, mode: 'edit' | 'create'): void => {
+      setFund(fund);
+      setFundModalMode(mode);
+      openModal(Modal.SAME);
+    },
+    [openModal],
+  );
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:86]: Modal should be opened only on certain actions, triggering modal opening on keyboard events doesn't make sense

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 4]
```diff
@@ -9,69 +1,89 @@
...
-    useState<boolean>(false);
-  const [fundUpdateModalIsOpen, setFundUpdateModalIsOpen] =
-    useState<boolean>(false);
 
-  const [taxDeductible, setTaxDeductible] = useState<boolean>(true);
-  const [isArchived, setIsArchived] = useState<boolean>(false);
-  const [isDefault, setIsDefault] = useState<boolean>(false);
+  if (!orgId) {
+    return <Navigate to={'/'} replace />;
+  }
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:58]: Adding fallback i.e. default value for orgId, not possible as its dynamic representing organizationId

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 5]
```diff
@@ -9,69 +1,89 @@
...
+    'createdAt_DESC',
+  );
+
+  const [modalState, setModalState] = useState<{ [key in Modal]: boolean }>({
+    [Modal.SAME]: false,
+    [Modal.DELETE]: false,
   });
+  const [fundModalMode, setFundModalMode] = useState<'edit' | 'create'>(
+    'create',
+  );
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:72]: Renaming Enum Modal to ModalState doesn't make any change, and is not required

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 6]
```diff
@@ -238,187 +137,250 @@
...
+      headerAlign: 'center',
+      sortable: false,
+      headerClassName: `${styles.tableHeader}`,
+      renderCell: (params: GridCellParams) => {
+        return (
+          <div
+            className="d-flex justify-content-center fw-bold"
+            data-testid="fundName"
+            onClick={() => handleClick(params.row._id as string)}
+          >
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:169]: Adding keyboard events is not good, as user may be directed to other screen if unaware of keyboard events

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 7]
```diff
@@ -238,187 +137,250 @@
...
+            variant="outline-success"
+            size="sm"
+            className="rounded"
+            onClick={() => handleClick(params.row._id as string)}
+            data-testid="viewBtn"
+          >
+            <i className="fa fa-eye me-1" />
+            {t('viewCampaigns')}
+          </Button>
+        );
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:275]: Adding keyboard events is not good, as user may be directed to other screen if unaware of keyboard events

File: src/screens/OrganizationFunds/OrganizationFunds.module.css [hunk 8]
```diff
@@ -49,12 +38,33 @@
...
+}
+
 .titlemodal {
+  color: #707070;
   font-weight: 600;
-  font-size: 25px;
-  margin-top: 1rem;
+  font-size: 32px;
   width: 65%;
+  margin-bottom: 0px;
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.module.css:52]: CSS styling could be handled in other separate PR

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 9]
```diff
@@ -9,69 +1,89 @@
...
-  tableCellClasses,
-} from '@mui/material';
+import { Stack } from '@mui/material';
 import dayjs from 'dayjs';
 import { FUND_LIST } from 'GraphQl/Queries/fundQueries';
+import {
+  DataGrid,
+  type GridCellParams,
+  type GridColDef,
+} from '@mui/x-data-grid';
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:18]: Done

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 10]
```diff
@@ -9,69 +1,89 @@
...
+  '& .MuiDataGrid-main': {
+    borderRadius: '0.5rem',
   },
-}));
+};
+
+enum Modal {
+  SAME = 'same',
+  DELETE = 'delete',
+}
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:45]: Done

File: src/screens/OrganizationFunds/OrganizationFunds.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:72]: Anyways done

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 11]
```diff
@@ -238,187 +137,250 @@
...
+      flex: 1,
+      minWidth: 100,
+      align: 'center',
+      headerAlign: 'center',
+      headerClassName: `${styles.tableHeader}`,
+      sortable: false,
+      renderCell: (params: GridCellParams) => {
+        return <div>{params.row.id}</div>;
+      },
+    },
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:153]: Adding keyboard events such as onKey will lead users being redirected unwantedly if unaware of such events

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 12]
```diff
@@ -9,69 +1,89 @@
...
+    backgroundColor: 'transparent',
+  },
+  '& .MuiDataGrid-root': {
+    borderRadius: '0.5rem',
+  },
+  '& .MuiDataGrid-main': {
+    borderRadius: '0.5rem',
   },
-}));
+};
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:40]: Not required

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 13]
```diff
@@ -238,187 +137,250 @@
...
+              variant="danger"
+              className="rounded"
+              data-testid="deleteFundBtn"
+              onClick={() => handleDeleteClick(params.row as InterfaceFundInfo)}
+            >
+              <i className="fa fa-trash" />
+            </Button>
+          </>
+        );
+      },
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:252]: Adding keyboard events such as onKey will lead users being redirected unwantedly if unaware of such events

File: src/screens/OrganizationFunds/FundDeleteModal.tsx [hunk 14]
```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/FundDeleteModal.tsx:2]: Folder containing mutation queries is named GraphQl not graphql

File: src/screens/OrganizationFunds/FundDeleteModal.tsx [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+        variables: {
+          id: fund?._id,
+        },
+      });
+      refetchFunds();
+      hide();
+      toast.success(t('fundDeleted'));
+    } catch (error: unknown) {
+      toast.error((error as Error).message);
+    }
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/FundDeleteModal.tsx:42]: The toast is displayed on delete Action, hence giving additional context of origin of error is not required. And error.message sent from server is enough for context

File: src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx [hunk 16]
```diff
@@ -60,230 +52,282 @@ const orgFundCampaign = (): JSX.Element => {
...
+  const [searchTerm, setSearchTerm] = useState('');
+  const [sortBy, setSortBy] = useState<string | null>(null);
+
+  const [modalState, setModalState] = useState<{ [key in Modal]: boolean }>({
+    [Modal.SAME]: false,
+    [Modal.DELETE]: false,
   });
+  const [campaignModalMode, setCampaignModalMode] = useState<'edit' | 'create'>(
+    'create',
+  );
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx:71]: Done

File: src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx [hunk 17]
```diff
@@ -60,230 +52,282 @@
...
+    setModalState((prevState) => ({ ...prevState, [modal]: false }));
+
+  const handleOpenModal = useCallback(
+    (campaign: InterfaceCampaignInfo | null, mode: 'edit' | 'create'): void => {
+      setCampaign(campaign);
+      setCampaignModalMode(mode);
+      openModal(Modal.SAME);
+    },
+    [openModal],
+  );
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx:85]: keyboard events not required

File: src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx [hunk 18]
```diff
@@ -9,49 +1,49 @@
...
+    backgroundColor: 'transparent',
+  },
+  '& .MuiDataGrid-root': {
+    borderRadius: '0.5rem',
   },
-}));
+  '& .MuiDataGrid-main': {
+    borderRadius: '0.5rem',
+  },
+};
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx:41]: not required

File: src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx [hunk 19]
```diff
@@ -60,230 +52,282 @@
...
+            variant="outline-success"
+            size="sm"
+            className="rounded"
+            data-testid="viewBtn"
+            onClick={() => handleClick(params.row.campaign._id as string)}
+          >
+            <i className="fa fa-eye me-1" />
+            {t('viewPledges')}
+          </Button>
+        );
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx:297]: keyboard events not required

File: src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx [hunk 20]
```diff
@@ -60,230 +52,282 @@
...
+      headerAlign: 'center',
+      headerClassName: `${styles.tableHeader}`,
+      sortable: false,
+      renderCell: (params: GridCellParams) => {
+        return (
+          <div
+            className="d-flex justify-content-center fw-bold"
+            data-testid="campaignName"
+            onClick={() => handleClick(params.row.campaign._id as string)}
+          >
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx:174]: keyboard events not required

File: src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx [hunk 21]
```diff
@@ -60,230 +52,282 @@
...
-    useState<boolean>(false);
-  const [campaignUpdateModalIsOpen, setcampaignUpdateModalIsOpen] =
-    useState<boolean>(false);
-  const [campaignDeleteModalIsOpen, setcampaignDeleteModalIsOpen] =
-    useState<boolean>(false);
+  const { fundId, orgId } = useParams();
+
+  if (!fundId || !orgId) {
+    return <Navigate to={'/'} />;
+  }
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx:59]: No need of adding fallback, as we are handling scenario where they could be undefined

File: src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx [hunk 22]
```diff
@@ -9,49 +1,49 @@
...
-  TableContainer,
-  TableHead,
-  TableRow,
-  styled,
-  tableCellClasses,
-} from '@mui/material';
+import type { GridCellParams, GridColDef } from '@mui/x-data-grid';
+import { DataGrid } from '@mui/x-data-grid';
+import { currencySymbols } from 'utils/currency';
+import { Stack, Typography, Breadcrumbs, Link } from '@mui/material';
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx:20]: Done

File: src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx [hunk 23]
```diff
@@ -9,49 +1,49 @@
...
-}));
+  '& .MuiDataGrid-main': {
+    borderRadius: '0.5rem',
+  },
+};
+
+enum Modal {
+  SAME = 'same',
+  DELETE = 'delete',
+}
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx:46]: Done

File: src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx [hunk 24]
```diff
@@ -60,230 +52,282 @@
...
+              variant="danger"
+              className="rounded"
+              data-testid="deleteCampaignBtn"
+              onClick={() =>
+                handleDeleteClick(params.row.campaign as InterfaceCampaignInfo)
+              }
+            >
+              <i className="fa fa-trash" />
+            </Button>
+          </>
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx:272]: keyboard events, not required

File: src/screens/FundCampaignPledge/FundCampaignPledge.module.css [hunk 25]
```diff
@@ -125,3 +126,76 @@
...
+  gap: 7rem;
+  width: 100%;
+  justify-content: space-between;
+  margin: 1.5rem 0 0 0;
+  padding: 1.25rem 2rem;
+  background-color: rgba(255, 255, 255, 0.591);
+
+  box-shadow: rgba(0, 0, 0, 0.16) 0px 1px 4px;
+  border-radius: 0.5rem;
+}
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/FundCampaignPledge/FundCampaignPledge.module.css:141]: Done

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 26]
```diff
@@ -238,187 +139,250 @@ const organizationFunds = (): JSX.Element => {
...
+      headerAlign: 'center',
+      sortable: false,
+      headerClassName: `${styles.tableHeader}`,
+      renderCell: (params: GridCellParams) => {
+        return (
+          <div
+            className="d-flex justify-content-center fw-bold"
+            data-testid="fundName"
+            onClick={() => handleClick(params.row._id as string)}
+          >
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:171]: Not required

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 27]
```diff
@@ -4,74 +1,91 @@
...
+    backgroundColor: 'transparent',
   },
-}));
+  '& .MuiDataGrid-root': {
+    borderRadius: '0.5rem',
+  },
+  '& .MuiDataGrid-main': {
+    borderRadius: '0.5rem',
+  },
+};
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:40]: Not required

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 28]
```diff
@@ -238,187 +139,250 @@
...
+              variant="danger"
+              className="rounded"
+              data-testid="deleteFundBtn"
+              onClick={() => handleDeleteClick(params.row as InterfaceFundInfo)}
+            >
+              <i className="fa fa-trash" />
+            </Button>
+          </>
+        );
+      },
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:254]: Not required

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 29]
```diff
@@ -238,187 +139,250 @@
...
+          <div>
+            <Button
+              variant="success"
+              onClick={() => handleOpenModal(null, 'create')}
+              style={{ marginTop: '11px' }}
+              data-testid="createFundBtn"
+            >
+              <i className={'fa fa-plus me-2'} />
+              {t('createFund')}
+            </Button>
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:342]: not required

File: src/screens/OrganizationFunds/OrganizationFunds.tsx [hunk 30]
```diff
@@ -4,74 +1,91 @@
...
+  },
+  '& .MuiDataGrid-main': {
+    borderRadius: '0.5rem',
+  },
+};
+
+enum ModalState {
+  SAME = 'same',
+  DELETE = 'delete',
+}
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFunds/OrganizationFunds.tsx:45]: not required

File: src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx [hunk 31]
```diff
@@ -1,57 +1,52 @@
...
-}));
+  '& .MuiDataGrid-main': {
+    borderRadius: '0.5rem',
+  },
+};
+
+enum ModalState {
+  SAME = 'same',
+  DELETE = 'delete',
+}
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/OrganizationFundCampaign/OrganizationFundCampagins.tsx:49]: not required

**Comment by GlenDsza** [comment]
@tasneemkoushar Added code to view additional volunteers in a scroll-enabled popup. Plz review

https://github.com/PalisadoesFoundation/talawa-admin/assets/109027247/2ddec728-4382-49ea-9b77-474b985bccb6

**Comment by GlenDsza** [comment]
> 1. The term `volunteers` is inappropriate as they are not volunteering to manually assist, but are making a financial pledge to a fund or campaign.
> 2. They are `pledgers`. Please make the update to the appropriate pledge screens.
> 3. Also, the amount raised isn't valid until funds are paid. The term `Raised` should be `Pledged`. We don't have a payment acceptance plugin yet.

@palisadoes Agree we don't have plugin for fund acceptance, but for now have kept it as constant and will add mock functionality of payments in user screen for pledges in [Link](https://github.com/PalisadoesFoundation/talawa-admin/issues/2070). So accordingly UI could be designed. Once we integrate payment plugins we can actually add a payment interface before paying.

**Comment by palisadoes** [comment]
- In that case shouldn't we have two chart lines on the dashboard? One for Raised and the other for Donated? 
- The Donated line could be blank for now so that we remember to fix it.

**Comment by GlenDsza** [comment]
@palisadoes adding 2 progress bar wont look good and make dashboard over crowded with similar components, would propose following options:
1. add a dropdown with down arrow on raised/pledged text in order to switch between raised and pledged. 
2. Show a group radio button to switch from pledged to raised 

image

**Comment by palisadoes** [comment]
Yes, that would be better

**Comment by tasneemkoushar** [comment]
@GlenDsza can you please share a clip of the new screen after making all the above changes mentioned in the comments by @palisadoes. 

1. Can you change the name from Volunteer to pledger.
2. Also in the below image what will be the view if we have around 5-10 pledgers. In the clip, can you try to add more than 5 pledgers and show it.
image

File: src/screens/FundCampaignPledge/PledgeModal.tsx [hunk 32]
```diff
@@ -190,31 +190,31 @@ const PledgeModal: React.FC<InterfacePledgeModal> = ({
...
                   setFormState({
                     ...formState,
-                    pledgeUsers: newVolunteers,
+                    pledgeUsers: newPledgers,
                   });
                 }
               }
               renderInput={(params) => (
-                <TextField {...params} label="Volunteers" />
+                <TextField {...params} label="Pledgers" />
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/FundCampaignPledge/PledgeModal.tsx:217]: pledgeUsers will always be an array either containing list of users associated with pledge or empty array, so indirectly  this scenario is handled.

File: src/screens/FundCampaignPledge/PledgeModal.tsx [hunk 33]
```diff
@@ -78,7 +78,7 @@ const PledgeModal: React.FC<InterfacePledgeModal> = ({
   useEffect(() => {
     if (memberData) {
       /*istanbul ignore next*/
-      setVolunteers(memberData.organizations[0].members);
+      setPledgers(memberData.organizations[0].members);
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/screens/FundCampaignPledge/PledgeModal.tsx:81]: before setting state variable I am already checking for existence of `memberData` and memberData.organizations can never be null as each member is linked to at least a default organization.

**Comment by GlenDsza** [comment]
- The scenario in which we have lots of pledgers, the popup is a scroll enabled container containing pledgers
- Changed volunteers to pledgers
- Added support to view progress bar for both pledged amount and raised amount

https://github.com/PalisadoesFoundation/talawa-admin/assets/109027247/ea81c213-c876-4523-8584-3a5a493d6c21

**Review by Nitya-Pasrija — COMMENTED** [review]
Looks good @GlenDsza 

Quick question: What happens when the goal is reached? Do we start mapping the excess?? I hope the site doesn't break cuz of slider excess

**Comment by GlenDsza** [comment]
> Looks good @GlenDsza
> 
> Quick question: What happens when the goal is reached? Do we start mapping the excess?? I hope the site doesn't break cuz of slider excess

@Nitya-Pasrija progress bar will show completely filled if amount crosses set goal.

image

In above image campaign goal was $3000 and the amount was $10000

**Comment by palisadoes** [comment]
@tasneemkoushar PTAL

**Review by tasneemkoushar — APPROVED** [review]

**Comment by tasneemkoushar** [comment]
@GlenDsza LGTM feel free to merge

**Review by tasneemkoushar — APPROVED** [review]

**Pull request merged**